src/lib/mina_lib/mina_lib.ml

```diff
@@ -1582,58 +1618,19 @@ let create ?wallets (config : Config.t) =
             (network_pipe, api_pipe, new_blocks_pipe)
           in
           trace_task "transaction pool broadcast loop" (fun () ->
-              let rl =
-                Network_pool.Rate_limiter.create
-                  ~capacity:
-                    ( Network_pool.Transaction_pool.Resource_pool.Diff
-                      .max_per_15_seconds
-                    , `Per (Time.Span.of_sec 15.) )
-              in
-              (* HACK: Pretend we're a remote peer so that we can rate limit
-                 ourselves.
-              *)
-              let us =
-                { Network_peer.Peer.host= Unix.Inet_addr.of_string "127.0.0.1"
-                ; libp2p_port= 0
-                ; peer_id= "" }
-              in
+              let rl = Network_pool.Transaction_pool.create_rate_limiter () in
+              log_rate_limiter_occasionally rl ~label:"local_transactions" ;
```

Comment:
Nit: this also applies to rebroadcast of non-local transactions; perhaps the message should be `transactions`?